### PR TITLE
temporary developer guide section for Budget

### PR DESCRIPTION
**Pull request by austinsantoso**
give brief description of Budget in the Developer Guide

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -554,6 +554,62 @@ Cell Formatting
...
+1. the monthly budget
+2. the current month
+3. the sum of all spending in the same month.
+
+All of which are saved with the `SpendingBook`.
+
+Upon initialization, if a save file is found it will automatically set the monthly budget based on the
+save file's options.
+
+Upon initialization MoneyGoWhere will check the current date. If the current month is different from the current month
```
Comment (Nanosync) [inline docs/DeveloperGuide.adoc]: Can combine with previous sentence

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -554,6 +554,62 @@ Cell Formatting
 ** """yummy"", ""juicy"""
 ** """fresh"", ""clean"""
 
+=== Budget Feature
+The `Budget` component has two core feature:
+
+1. Allowing the user to set the monthly budget
+2. Calculating the remanding budget after all the spending in the same month.
+
+The budget component allows the users to modify the current monthly budget `budget AMOUNT` command.
```
Comment (Nanosync) [inline docs/DeveloperGuide.adoc]: what is AMOUNT? the input scope should be defined

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -554,6 +554,62 @@ Cell Formatting
...
+The figure above shows the sequence of events that occur to set the monthly budget.
+The `LogicManager` receives the user input and passes it to the `SpendingBookParser` to parse the user input,
+resulting in a `BudgetCommand`.
+If the user input is valid, the command is then executed to access the Budget in Model and modify the value.
+
+==== Design Consideration
+
+[width="90%", cols="1,2,2", options="header",]
+|===
+|Aspect |Alternative 1 |Alternative 2
```
Comment (Nanosync) [inline docs/DeveloperGuide.adoc:605]: Swap alternative 1 and 2. Alternative 1 is the selected choice.

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -554,6 +554,62 @@ Cell Formatting
...
+the sum from the budget.
+
+==== Current Implementation
+The current `Budget` component keeps track of three variables:
+
+1. the monthly budget
+2. the current month
+3. the sum of all spending in the same month.
+
+All of which are saved with the `SpendingBook`.
```
Comment (Nanosync) [inline docs/DeveloperGuide.adoc]: within the SpendingBook

**Review by Nanosync — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.adoc (hunk 2 above)
Comment (austinsantoso) [reply docs/DeveloperGuide.adoc]: so something like, `budget amount` command, where amount is the desired monthly budget?

Comment (Nanosync) [reply docs/DeveloperGuide.adoc]: Is amount an integer, etc

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -560,6 +560,61 @@ Alternative 1 was chosen due to the ease of implementation.
 As the current statistics feature does not require a large amount of data, the number of method calls is not too large.
 Alternative 2 requires a change in the architecture which may result in higher complexity.
 
+=== Budget Feature
+The `Budget` component has two core feature:
```
Comment (Nanosync) [inline docs/DeveloperGuide.adoc]: Features

**Review by Nanosync — APPROVED** [review]

**Pull request merged**